qcodes/interactive_widget.py

```diff
@@ -0,0 +1,436 @@
+                dct
+            )  # TODO: include and extract more data!
+            text_input = Textarea(
+                value=description,
+                placeholder="Enter text",
+                disabled=True,
+                layout=Layout(height="300px", width="auto"),
+            )
+            plot_button = button(
+                "Plot",
+                "warning",
+                on_click=_do_in_tab(tab, ds, "plot"),
+                button_kwargs=dict(icon="line-chart"),
+            )
+            snapshot_button = button(
+                "Open snapshot",
+                "warning",
+                on_click=_do_in_tab(tab, ds, "snapshot"),
+                button_kwargs=dict(icon="camera"),
+            )
+            back_button = button(
+                "Back",
+                "warning",
+                on_click=_input_to_button(dct, box),
+                button_kwargs=dict(icon="undo"),
+            )
+            box.children = (
+                text_input,
+                snapshot_button,
+                plot_button,
+                back_button,
+            )
+
+        return on_click
+
+    def _input_to_button(dct: Dict, box: Box) -> Callable[[Button], None]:
+        def on_click(_: Button) -> None:
+            box.children = (_changeable_button(dct, box),)
+
+        return on_click
+
+    def _changeable_button(dct: Dict, box: Box) -> Button:
+        return button(
+            ", ".join(dct), "success", on_click=_button_to_input(dct, box),
+        )
+
+    box = VBox([], layout=Layout(height="auto", width="auto"))
+    box.children = (_changeable_button(dct, box),)
+    return box
+
+
+def _get_coords_and_vars(ds: DataSet) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    coordinates = {}
+    variables = {}
+    for p, spec in ds.paramspecs.items():
+        attrs = {
+            "unit": spec.unit,
+            "label": spec.label,
+            "type": spec.type,
+        }
+        if spec.depends_on:
+            attrs["depends_on"] = spec.depends_on.split(", ")  # type: ignore
+            variables[p] = attrs
+        else:
+            coordinates[p] = attrs
+    return coordinates, variables
+
+
+def _experiment_widget(tab: Tab) -> GridspecLayout:
+    """Show a `ipywidgets.GridspecLayout` with information about the
+    loaded experiment. The clickable buttons can perform an action in ``tab``.
+    """
+    header_names = [
+        "Run ID",
+        "Name",
+        "Experiment",
+        "Coordinates",
+        "Variables",
+        "MSMT Time",
+        "Notes",
+    ]
+
+    header = {n: button(n, "info") for n in header_names}
+    rows = [header]
+    for exp in qcodes.experiments():
+        tooltip = f"{exp.name}#{exp.sample_name}@{exp.path_to_db}"
+
+        for ds in exp.data_sets():
+            coords, variables = _get_coords_and_vars(ds)
+            row = {}
+            row["Run ID"] = label(str(ds.run_id))
+            row["Name"] = label(ds.name)
```

Comment:
sometimes these names are long, and the widget truncates them with "..." at the end - do ipywidgets have a different way for displaying such that the full name is still possible to see? perhaps a hover reveals full name, or it's wraped or else?

Reply:
In f8aad59d4, I add DataSet.name to the expanded Experiment button text area for the case where the Name col is too long.